hadoop-ozone/common/src/test/java/org/apache/hadoop/ozone/om/lock/TestOzoneManagerLock.java

```diff
@@ -115,6 +118,26 @@ void testLockingOrder() {
     }
   }
 
+  @ParameterizedTest
+  @EnumSource
+  public void testFlatLockWithParallelResource(FlatResource flatResource) {
+    OzoneManagerLock lock = new OzoneManagerLock(new OzoneConfiguration());
+    List<Resource> resources = new ArrayList<>();
+    resources.addAll(Arrays.stream(LeveledResource.values()).collect(Collectors.toList()));
+    resources.addAll(Arrays.stream(FlatResource.values()).collect(Collectors.toList()));
+    for (Resource otherResource : resources) {
+      String[] otherResourceName = generateResourceName(otherResource);
+      String[] flatResourceName = generateResourceName(flatResource);
+      lock.acquireWriteLock(otherResource, otherResourceName);
+      try {
+        lock.acquireWriteLock(flatResource, flatResourceName);
```

Comment:
Let's add a `@Timeout` annotation to this case. In the negative case where `flatResource` is actually contentious with `otherResource`, it'd become a deadlock.